### PR TITLE
Controlled U3 with non-trivial global phase fix

### PR DESCRIPTION
**Pull request by positr0nium**
## Description

A bug prevented the proper compilation of **controlled `U3Gate`s with non-trivial global phases** — the unitary returned by the controlled operation did not match the unitary of its transpiled definition circuit.

This PR adds a fix (the controlled U3 synthesis now correctly accounts for `global_phase` in the A–B–C decomposition) and a comprehensive 89‑test suite for `U3Gate`.

## Related Issues

This fixes a previously unknown issue.

## Type of Change

- [x] Bug Fix
- [ ] Feature (new functionality)
- [ ] Change Request (modification of existing functionality)
- [ ] Refactoring (no behavior change)
- [ ] Performance improvement
- [ ] Documentation
- [ ] CI / Build

## Breaking Change?
- [ ] Yes
- [x] No

## What was changed?

- **`src/qrisp/circuit/controlled_operations.py`** — `multi_controlled_u3_circ`: the A–B–C decomposition of a general U3 gate (Barenco et al., 1995) now correctly accounts for a non-zero `global_phase`.
- **`tests/circuit_tests/operation_tests/test_u3_gate.py`** — 89 tests across 11 test classes covering initialization, unitary, inverse, controlled & multi-controlled gates, named variants, abstract params, PauliGate subclass, transpile roundtrips, and edge cases.

## How was it tested?

Reproduction of the bug (before fix):

```python
from qrisp import *
from numpy.linalg import norm

def check_params(a, b, c, d):
    gate = U3Gate(a, b, c, global_phase=d)
    cgate = gate.control(1)
    assert norm(cgate.get_unitary() - cgate.definition.transpile().get_unitary()) < 1e-4

check_params(2.5740044828568522, 3.141592653589793, 3.141592653589793, 3.141592653589793)  # ❌ before fix
check_params(2.5740044828568522, 3.141592653589793, 3.141592653589793, 0)                    # ❌
check_params(2.5740044828568522, 0, 0, 0)                                                     # ❌
check_params(2.5740044828568522, 0, 0, 1)                                                     # ❌
check_params(1, 2, 3, 4)                                                                      # ❌
```

All 89 tests pass after the fix:

```
89 passed, 1 warning in 10.32s
```

| Test-ID | Status |
|---------|--------|
| T-001 – Controlled U3 with non-trivial global phase (bug repro) | ✅ |
| T-002 – Multi-controlled U3 (2‑ctrl, 3‑ctrl) | ✅ |
| T-003 – Named variant controlled gates (P, RX, RY, RZ) | ✅ |
| T-004 – Global-phase-only identity gate controlled | ✅ |

## Screenshots / Output (if applicable)

```
(pytest output)

89 passed, 1 warning in 10.32s
```

## Checklist
- [x] My code follows the project's coding standards
- [x] I have performed a self-review
- [x] I have added/updated tests (referencing issue Test-IDs)
- [x] All tests pass locally and in CI
- [x] I have updated the documentation
- [x] I have added a changelog entry (not necessary)
- [x] Breaking changes are documented with migration path (not necessary)

## Reviewer Notes

- The `ControlledOperation.get_unitary()` method provides the reference unitary via direct matrix construction (`controlled_unitary`), while `definition.transpile().get_unitary()` exercises the full decomposition→transpile pipeline.
- All named-variant tests use the standard gate factory functions from `standard_operations.py` (`PGate`, `RZGate`, `RXGate`, `RYGate`, `HGate`, `SGate`, `TGate`, `GPhaseGate`).

### CODE REVIEW

File: src/qrisp/alg_primitives/mcx_algs/gray_pt.py [hunk 1]
```diff
@@ -60,20 +60,4 @@ def pt_multi_cx(n, reduced=False):
 
         res = input_qv.qs.copy()
 
-    return res.to_gate(f"pt{n}cx")
-
-
-def gray_pt_mcx(n, ctrl_state):
```
Comment (renezander90) [inline src/qrisp/alg_primitives/mcx_algs/gray_pt.py:66]: Why was this function removed?

Comment (positr0nium) [reply src/qrisp/alg_primitives/mcx_algs/gray_pt.py:66]: It was an internal function that was called in only two locations and removing it contributes to making the mcx compilation infrastructure less bloated.

File: src/qrisp/circuit/operation.py [hunk 2]
```diff
@@ -96,7 +96,7 @@ def __init__(
 
         elif not isinstance(name, str):
             raise Exception(
-                "Tried to create a Operation with name of type({type(name)} (required is str)"
+                f"Tried to create a Operation with name of type({type(name)} (required is str)"
             )
```
Comment (renezander90) [inline src/qrisp/circuit/operation.py:100]:
```suggestion
if not isinstance(name, str):
    raise TypeError(
        f"Tried to create an Operation with name of type {type(name).__name__} "
        "(required is str)."
    )
```

**Review by renezander90 — APPROVED** [review]
